x/auth/ante.go

```diff
@@ -160,14 +159,14 @@ func processSig(
 
 	// Check account number.
 	accnum := acc.GetAccountNumber()
-	if accnum != sig.AccountNumber {
+	if !simulate && (accnum != sig.AccountNumber) {
 		return nil, sdk.ErrInvalidSequence(
 			fmt.Sprintf("Invalid account number. Got %d, expected %d", sig.AccountNumber, accnum)).Result()
 	}
 
 	// Check and increment sequence number.
 	seq := acc.GetSequence()
-	if seq != sig.Sequence {
+	if !simulate && (seq != sig.Sequence) {
```

Comment:
I'm confused, you should still have the correct sequence number when simulating a tx?

Reply:
When simulating, `sig` would be empty - line 68, `sigs := stdTx.GetSignatures()`, `sigs` would just be a 0-length slice.

Reply:
Makes sense, can you add a comment here explaining that?